sdk/src/main/scala/com/horizen/certificatesubmitter/CertificateSubmitter.scala

```diff
@@ -200,52 +200,69 @@ class CertificateSubmitter
     }
   }
 
-  private def getCertificateTopQuality(): Int = {
+  private def getCertificateTopQuality(epoch: Int): Try[Long] = Try {
     mainchainChannel.getTopQualityCertificates(BytesUtils.toHexString(BytesUtils.reverseBytes(params.sidechainId))) match {
       case Success(topQualityCertificates)=>
-        topQualityCertificates.mempoolCertInfo.quality.getOrElse(topQualityCertificates.chainCertInfo.quality.getOrElse(0))
-      case Failure(_) => 0
+        if (!topQualityCertificates.mempoolCertInfo.quality.isEmpty) {
```

Comment:
I would suggest to add the same condition as for inchain cert: ` && topQualityCertificates.mempoolCertInfo.epoch.get == epoch`